### DISCUSSION TITLE
Transparently work with `*.11tydata.*` filenames in Global data files

### DISCUSSION THREAD

**Discussion by gloryofrobots** [11ty/eleventy#4071] [Enhancement Queue]
**Describe the bug**

global data with 11tydata.json extension does not work due to the problem with TemplateData.getObjectPathForDataFile(); it returns ```filename.11tydata``` as a path for lodashset
instead of ```filename```

**To Reproduce**

create file test.11tydata.json in _data dir
with 
```
{
    "key":"value"
}
```
add  ```<h1> {{test.key}} </h1> ``` to your template
**Expected behavior**
```<h1>value</h1> ```must be rendered instead of ```<h1></h1>```

**Additional context**
I am working on PR for #794 and can fix this issue too. Or maybe this is correct behavior?
Please let me know

**Comment by jevets** [comment]
This is the behavior I'd expect since it's `json` and not `js`. 

I'd name the file `test.json`. I've only ever used `11tydata.` extension prefix for `.js` (and `.11ty` for `.js` template files).

Seeing `_data/test.11tydata.json`, I'd expect Eleventy to name that data `test.11tydata`, so I'd access `key` as `{{ test.11tydata.key }}` in templates.

For a JavaScript data file named `_data/test.11tydata.js`, I'd access as `{{ test.key }}` in templates.

**Comment by gloryofrobots** [comment]
```11tydata.js``` or  ```11tydata.json``` for global data it doesn`t matter. Global data object from 
 ```_data/test.11tydata.js``` will be accessible as **{{test.11tydata.key}}**  not **{{test.key}}** I just tested it.

**Comment by jevets** [comment]
[Global data files](https://www.11ty.dev/docs/data-global/) don't expect the `.11tydata` extension prefixes. Eleventy doesn't remove the strings of the file name when naming the data object; it assumes you wanted the resulting data object to be named `test.11tydata` instead of `test`.

Examples:

- A `key` in a global data file named `_data/one.js` would be available as `{{ one.key }}` in templates.
- A `key` in a global data file named `_data/one.two.js` would be available as `{{ one.two.key }}` within templates.
- A `key` in a global data file named `_data/one/two.js` would also be available as `{{ one.two.key }}` from within templates. (may be a conflict there, not sure)
- A `key` in a global data file named `_data/one.two.three.four.js` would be available as `{{ one.two.three.four.key }}` within templates.

But [Template/Directory data files](https://www.11ty.dev/docs/data-template-dir/) **do expect** the `.11tydata` prefixes, to differentiate them from other files in the directory and to instruct Eleventy to parse them as (local) data files. And Eleventy does remove the special prefix from the data object's name.

See also [the config option to change the `jsDataFileSuffix`](https://www.11ty.dev/docs/config/#change-file-suffix-for-template-and-directory-data-files). It's only for template/directory data files.

(I called it "extension prefix" above; should've called it suffix.)

**Comment by zachleat** [comment]
For the record I do consider this a cognitive overhead upgrade if we support `.11tydata` in global data files just to maintain consistency between global data and template/dir data 👍

One less inconsistency to learn

**Comment by zachleat** [comment]
We _may_ want to separate these out into two separate PRs though (re: #821).

> Global data object from _data/test.11tydata.js will be accessible as {{test.11tydata.key}} not {{test.key}} I just tested it.

If someone has a `_data/test.11tydata.js` in place and “made it work” with the key `test.11tydata.key`, I wouldn’t want to break that working code.

So maybe global `.11tydata` might be a major version change

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
